CONTRIBUTING.md

```diff
@@ -3,11 +3,112 @@ Guidelines for Contributing
+- Smaller additions like individual functions can be merged directly after [@andralex](https://github.com/andralex) approves
+- Refactoring or bug fixes just need approval (LGTM) from two reviewers and enough time passed (usually 2-3 days, depending on weekends and holidays) to give everyone an opportunity to shout
+- Trivial changes (e.g. small documentation enhancements) are usually merged in less than 72h - otherwise ping for reviewers
+- If your PR is stalled (no activity for 2-3 days) and you addressed all concerns, don't hesitate to ping your reviewers
+- Ping [@9il](https://github.com/9il) for anything math related
+
+### Naming
+
+- [ ] Do you use clear variable names?
+- [ ] Does your PR have a clear and informative title?
+- [ ] Are your Git messages clear and informative?
+
+### Fixing issues
+
+- Bug fix PRs should have a separate commit for each bugzilla issue that it fixes
+- The message of the commit actually fixes the bug should start with "Fix issue X" where X is the number of the corresponding [bugzilla](https://issues.dlang.org) issue. For example:
+
+```
++ Refactor the code assist with the bug fix
++ Fix issue 17000 - map doesn't accept some range types
++ Fix issue 17001 - map doesn't work with some other range type
+```
+
+- Unrelated fixes, refactorings or additions should preferably go separate pull request(s) (tightly related fixes are okay in the same PR)
+
+Documentation
+-------------
+
+### Documentation style
+
+- Use complete English sentences with correct syntax, grammar, and punctuation
+- Use precise technical terms. Avoids colloquialisms
+- Honestly describe limitations and dangers
+- Add `Params`, `Returns` and if needed `See_Also` sections
+- Provide at least one ddoced unittest for a new method
+- Have a brief one sentence summary for at the beginning of a method
+- Give a synopsis at the beginning of a module accompanied by an example
+
+### Documentation checks
+
+- [ ] Read your documentation text aloud - is it understandable?
+- [ ] Check the output of CyberShadows's DAutoTest of the documentation (or run [it](https://github.com/D-Programming-Language/dlang.org) yourself)
+- [ ] All user impacting changes should have a corresponding [changelog](https://github.com/D-Programming-Language/phobos/blob/master/changelog.dd) entry
+- [ ] Did you add your method to the summary booktable or cheatsheat of a module? (might be in `package.d`)
+
+### Documentation tricks
+
+- Use backticks `` `code` `` instead of `$(D foo)`
+- Start with a Capital letter in all blocks of the method header
+- Use `$(REF)` for links to other methods (format is `$(REF _stripLeft, std, algorithm, mutation)`)
+- Use `$(REF_ALTTEXT)` for a different link name (format is `$(REF_ALTTEXT your text, formattedRead, std, format)`
+- Variables will be automatically put in backticks, use an underscore to avoid this behavior (e.g. `_foo`)
```

Comment:
Also, if you need to make a section heading with multiple words, you need to connect them with underscores instead of spaces, otherwise ddoc won't understand that it's a section heading:
````
/**
  This is a paragraph that ends with a colon:

  blah blah blah

  This_is_a_section_heading:

  blah blah blah
*/
````